tests/contrib/awards/test_awards_api.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Vocabularies is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Awards API tests."""
+
+from functools import partial
+
+import pytest
+from invenio_indexer.api import RecordIndexer
+from invenio_search import current_search_client
+from jsonschema import ValidationError as SchemaValidationError
+
+from invenio_vocabularies.contrib.awards.api import Award
+
+
+@pytest.fixture()
+def search_get():
+    """Get a document from an index."""
+    return partial(
+        current_search_client.get, Award.index._name, doc_type="_doc"
+    )
+
+
+@pytest.fixture()
+def indexer():
+    """Indexer instance with correct Record class."""
+    return RecordIndexer(
+        record_cls=Award,
+        record_to_index=lambda r: (r.__class__.index._name, "_doc"),
+    )
+
+
+@pytest.fixture()
+def example_award(db, award_full_data):
+    """Example award."""
+    awa = Award.create(award_full_data)
+    Award.pid.create(awa)
+    awa.commit()
+    db.session.commit()
+    return awa
+
+
+def test_award_schema_validation(app, db, example_award):
+    """Award schema validation."""
+    # valid data
+    awa = example_award
+
+    assert awa.schema == "local://awards/award-v1.0.0.json"
+    assert awa.pid
+    assert awa.id
+
+    # invalid data
+    examples = [
```

Comment:
minor: maybe we should ask @slint @ctrl-alt-soneca for some real life example of awards so we can abstract data properly. Just wondering if someone in the future sees this tests the data looks more like affiliations than actual awards (could be ok for funders tho...)